### PR TITLE
Expose WiFiStationProvision methods in ESP32Utils

### LINKED ISSUE #1599: Expose WiFiStationProvision methods in ESP32Utils
Expose WiFiStationProvision methods in ESP32Utils

- SetWiFiStationProvision
- ClearWiFiStationProvision
- GetWiFiStationProvision

### PR DESCRIPTION
**Pull request by shana-apple**
#### Problem
We need to expose wifi station provisioning methods in ESP32Utils

 #### Summary of Changes
Move (and modify accordingly) methods from NetworkProvisionningServerImpl to ESP32Utils

fixes #1599

### CODE REVIEW

File: src/platform/ESP32/ESP32Utils.cpp [hunk 1]
```diff
@@ -237,3 +238,70 @@ bool ESP32Utils::HasIPv6LinkLocalAddress(tcpip_adapter_if_t intfId)
...
+    wifi_config_t stationConfig;
+
+    err = esp_wifi_get_config(ESP_IF_WIFI_STA, &stationConfig);
+    SuccessOrExit(err);
+
+    VerifyOrExit(stationConfig.sta.ssid[0] != 0, err = CHIP_ERROR_INCORRECT_STATE);
+
+    netInfo.NetworkId              = kWiFiStationNetworkId;
+    netInfo.FieldPresent.NetworkId = true;
+    memcpy(netInfo.WiFiSSID, stationConfig.sta.ssid, min(strlen((char *) stationConfig.sta.ssid) + 1, sizeof(netInfo.WiFiSSID)));
```
Comment (andy31415) [inline src/platform/ESP32/ESP32Utils.cpp:254]: nit: should we use C++-style cast here? also should it be const char * ?

Should we ensure a null terminator even when ssid length overflows and sizeof is used?

File: src/platform/ESP32/ESP32Utils.cpp [hunk 2]
```diff
@@ -237,3 +238,70 @@ bool ESP32Utils::HasIPv6LinkLocalAddress(tcpip_adapter_if_t intfId)
...
+    wifi_config_t wifiConfig;
+
+    // Ensure that ESP station mode is enabled.  This is required before esp_wifi_set_config(ESP_IF_WIFI_STA,...)
+    // can be called.
+    err = ESP32Utils::EnableStationMode();
+    SuccessOrExit(err);
+
+    // Initialize an ESP wifi_config_t structure based on the new provision information.
+    memset(&wifiConfig, 0, sizeof(wifiConfig));
+    memcpy(wifiConfig.sta.ssid, netInfo.WiFiSSID, min(strlen(netInfo.WiFiSSID) + 1, sizeof(wifiConfig.sta.ssid)));
```
Comment (andy31415) [inline src/platform/ESP32/ESP32Utils.cpp:278]: should we ensure null terminator here and on the password?

**Review by kedars — APPROVED** [review]

File: src/platform/ESP32/ESP32Utils.cpp (hunk 2 above)
Comment (mspang) [reply src/platform/ESP32/ESP32Utils.cpp:278]: It seems to be documented as not having a null terminator (and in weave as well).

**Review by mspang — APPROVED** [review]

File: src/platform/ESP32/ESP32Utils.cpp (hunk 1 above)
Comment (rwalker-apple) [reply src/platform/ESP32/ESP32Utils.cpp:254]: +1

**Review by BroderickCarlin — APPROVED** [review]

File: src/platform/ESP32/ESP32Utils.cpp (hunk 2 above)
Comment (shana-apple) [reply src/platform/ESP32/ESP32Utils.cpp:278]: Yes this is actually Weave code in CHIP

File: src/platform/ESP32/ESP32Utils.cpp (hunk 1 above)
Comment (woody-apple) [reply src/platform/ESP32/ESP32Utils.cpp:254]: @shana-apple ?

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@shana-apple needs restyle merge too

**Review by jelderton — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Pull request merged**